### ISSUE TITLE
Multiple keys not working 

### ISSUE THREAD

**Issue by florin-lazau** [webfactory/ssh-agent#178]
I have a React project that has two private repositories as dependencies. I created two SSH keys, added private keys as secrets and added public keys as deploy keys in the dependency repository. 

This is the code from the pipeline 
```
      - name: SSH
        uses: webfactory/ssh-agent@v0.7.0
        with:
          ssh-private-key: |
            ${{ secrets.SSH_PRIVATE_KEY1 }}
            ${{ secrets.SSH_PRIVATE_KEY2 }}
```

For some reason, the connection works only for the first key. I have tried putting SSH_PRIVATE_KEY2 first, and it worked for the repository that has SSH_PRIVATE_KEY2 public key counterpart, then I put SSH_PRIVATE_KEY1 first, and it worked for the repository that has SSH_PRIVATE_KEY1 public key counterpart. 

Is someone else having the same issue?

**Comment by tompiler** [comment]
Yep, also having no luck. Even after following steps [here](https://github.com/webfactory/ssh-agent#using-multiple-deploy-keys-inside-docker-builds) to align git config and ssh config. I can clearly see by executing:

`RUN --mount=type=ssh GIT_SSH_COMMAND="ssh -v" pip install git+ssh://git@github.com/owner/repo-b`

that it is still trying to use the public deploy key from repo-a to authenticate to github for repository b. Really frustrating - been stuck on this for literally days and I think I'm as far as I can go.

**Comment by shaunco** [comment]
Just FYI, we moved the shaunco fork to https://github.com/mapped/ssh-agent/tree/git-repo-mapping

**Comment by de-flandres** [comment]
Ended up with working solution with multiple keys for different repositories. 
e.g.

Imagine we have three different github repositories (`A`, `B`):
- `github.com/myuser/A-repo1`
- `github.com/myuser/A-repo2`
- `github.com/myuser/B-repo1`

We also want this `C` repository to be able to access these three repos above:
- `github.com/myuser/C-repo1`

To be able to grant access `C-repo1` to them I've generated two ssh keys:
- 1st key for both `A-repo1` and `A-repo2`:
```
ssh-keygen -t ed25519 -C github.com/myuser/A-*
```
- 2nd key for `B-repo1`:
```
ssh-keygen -t ed25519 -C github.com/myuser/B-repo1
```

Then:
- saved both `public` keys under `myuser` github user settings (user needs to have access to `A-repo1`, `A-repo2`, `B-repo1`)
- saved both `private` keys under `C-repo1` actions secrets variables (repo settings) and named them `REPOS_A_PRIVATE_KEY ` and `REPO_B_PRIVATE_KEY ` respectively
- in `C-repo1` github actions CI workflow added a step:
```
- name: Setup ssh agent
  uses: webfactory/ssh-agent@v0.9.0
  with:
    ssh-private-key: |
      ${{ secrets.REPOS_A_PRIVATE_KEY }}
      ${{ secrets.REPO_B_PRIVATE_KEY }}
```

Now `C-repo1` has access to _all_ `A-*` repos and _specific_ `B-repo1`!

Not an ideal solution, but it works. Hopefully this will help someone.
